crates/oxc_linter/src/rules/jsx_a11y/no_noninteractive_tabindex.rs

```diff
@@ -0,0 +1,210 @@
+        .with_help("tabIndex attribute should be removed")
+        .with_label(span)
+}
+
+#[derive(Debug, Clone)]
+pub struct NoNoninteractiveTabindex(Box<NoNoninteractiveTabindexConfig>);
+
+#[derive(Debug, Clone)]
+struct NoNoninteractiveTabindexConfig {
+    tags: Vec<CompactStr>,
+    roles: Vec<CompactStr>,
+    allow_expression_values: bool,
+}
+
+impl Default for NoNoninteractiveTabindex {
+    fn default() -> Self {
+        Self(Box::new(NoNoninteractiveTabindexConfig {
+            roles: vec![CompactStr::new("tabpanel")],
+            allow_expression_values: true,
+            tags: vec![],
+        }))
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule checks that non-interactive elements don't have a tabIndex which would make them interactive via keyboard navigation.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Tab key navigation should be limited to elements on the page that can be interacted with.
+    /// Thus it is not necessary to add a tabindex to items in an unordered list, for example,
+    /// to make them navigable through assistive technology.
+    ///
+    /// These applications already afford page traversal mechanisms based on the HTML of the page.
+    /// Generally, we should try to reduce the size of the page's tab ring rather than increasing it.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```jsx
+    /// <div tabIndex="0" />
+    /// <div role="article" tabIndex="0" />
+    /// <article tabIndex="0" />
+    /// <article tabIndex={0} />
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```jsx
+    /// <div />
+    /// <MyButton tabIndex={0} />
+    /// <button />
+    /// <button tabIndex="0" />
+    /// <button tabIndex={0} />
+    /// <div />
+    /// <div tabIndex="-1" />
+    /// <div role="button" tabIndex="0" />
+    /// <div role="article" tabIndex="-1" />
+    /// <article tabIndex="-1" />
+    /// ```
+    NoNoninteractiveTabindex,
+    correctness,
+);
+
+const INTERACTIVE_HTML_ELEMENTS: phf::set::Set<&'static str> = phf_set! {
+    "a", "audio", "button", "details", "embed", "iframe", "img", "input", "label", "select", "textarea", "video"
+};
+
+const INTERACTIVE_HTML_ROLES: phf::set::Set<&'static str> = phf_set! {
+    "button", "checkbox", "gridcell", "link", "menuitem", "menuitemcheckbox", "menuitemradio", "option", "progressbar", "radio", "textbox"
+};
+
+impl Rule for NoNoninteractiveTabindex {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::JSXOpeningElement(jsx_el) = node.kind() else {
+            return;
+        };
+
+        if let Some(JSXAttributeItem::Attribute(tabindex_attr)) =
+            has_jsx_prop_ignore_case(jsx_el, "tabIndex")
+        {
+            if let Some(JSXAttributeValue::StringLiteral(tabindex)) = &tabindex_attr.value {
```

Comment:
lets flatten these into let - else stmts
```suggestion
        let Some(JSXAttributeItem::Attribute(tabindex_attr)) = has_jsx_prop_ignore_case(jsx_el, "tabIndex") else { return; };
        let Some(JSXAttributeValue::StringLiteral(tabindex)) = &tabindex_attr.value else { return; };
```

as it's currently much more nested than it needs to be

Reply:
done